src/build/inject-babel-helpers.ts

```diff
@@ -0,0 +1,63 @@
+import * as dom5 from 'dom5';
+import {fs} from 'mz';
+import * as parse5 from 'parse5';
+import * as path from 'path';
+
+import {AsyncTransformStream, getFileContents} from './streams';
+
+import File = require('vinyl');
+
+const p = dom5.predicates;
+
+const scriptOrImport = p.OR(
+    p.hasTagName('script'),
+    p.AND(p.hasTagName('link'), p.hasSpaceSeparatedAttrValue('rel', 'import')));
+
+
+
+/**
+ * When compiling to ES5 we need to inject Babel's helpers into a global so
+ * that they don't need to be included with each compiled file.
+ */
+export class BabelHelpersInjector extends AsyncTransformStream<File, File> {
+  constructor(private entrypoint: string) {
+    super({objectMode: true});
+  }
+
+  protected async * _transformIter(files: AsyncIterable<File>) {
+    for
+      await(const file of files) {
```

Comment:
clang-format misformats `for await` pretty bad :/

I was able to keep the two uses of it very short here to compensate.